tests/functional/ephemeral/test_ephemeral.py

```diff
@@ -0,0 +1,207 @@
+            "base_errors.sql": ephemeral_errors__base__base_sql,
+            "base_copy_errors.sql": ephemeral_errors__base__base_copy_sql,
+        },
+    }
+
+
+@pytest.fixture(scope="class", autouse=True)
+def setUp(project):
+    project.run_sql_file(project.test_data_dir / Path("seed.sql"))
+
+
+class TestEphemeralMulti:
+    def test_ephemeral_multi(self, project):
+        results = run_dbt(["run"])
+        assert len(results) == 3
+
+        check_relations_equal(project.adapter, ["seed", "dependent"])
+        check_relations_equal(project.adapter, ["seed", "double_dependent"])
+        check_relations_equal(project.adapter, ["seed", "super_dependent"])
+        assert os.path.exists("./target/run/test/models/double_dependent.sql")
+        with open("./target/run/test/models/double_dependent.sql", "r") as fp:
+            sql_file = fp.read()
+
+        sql_file = re.sub(r"\d+", "", sql_file)
+        expected_sql = (
+            'create view "dbt"."test_ephemeral"."double_dependent__dbt_tmp" as ('
+            "with __dbt__cte__base as ("
+            "select * from test_ephemeral.seed"
+            "),  __dbt__cte__base_copy as ("
+            "select * from __dbt__cte__base"
+            ")-- base_copy just pulls from base. Make sure the listed"
+            "-- graph of CTEs all share the same dbt_cte__base cte"
+            "select * from __dbt__cte__base where gender = 'Male'"
+            "union all"
+            "select * from __dbt__cte__base_copy where gender = 'Female'"
+            ");"
+        )
+        sql_file = "".join(sql_file.split())
+        expected_sql = "".join(expected_sql.split())
+        assert sql_file == expected_sql
+
+
+class TestEphemeralNested:
+    def test_ephemeral_nested(self, project):
+        results = run_dbt(
+            ["run", "--models", "ephemeral_level_two", "root_view", "ephemeral", "source_table"]
+        )
+        assert len(results) == 2
+        assert os.path.exists("./target/run/test/models/root_view.sql")
+        with open("./target/run/test/models/root_view.sql", "r") as fp:
+            sql_file = fp.read()
+
+        sql_file = re.sub(r"\d+", "", sql_file)
+        expected_sql = (
+            'create view "dbt"."test_ephemeral"."root_view__dbt_tmp" as ('
+            "with __dbt__cte__ephemeral_level_two as ("
+            'select * from "dbt"."test_ephemeral"."source_table"'
+            "),  __dbt__cte__ephemeral as ("
+            "select * from __dbt__cte__ephemeral_level_two"
+            ")select * from __dbt__cte__ephemeral"
+            ");"
+        )
+
+        sql_file = "".join(sql_file.split())
+        expected_sql = "".join(expected_sql.split())
+        assert sql_file == expected_sql
+
+
+class TestEphemeralErrorHandling:
+    def test_ephemeral_error_handling(self, project):
+        results = run_dbt(["run", "--models", "dependent_errors"], expect_pass=False)
+        assert len(results) == 1
+        assert results[0].status == "skipped"
```

Comment:
@ChenyuLInx any idea why this might showing incorrect status of `error`  in the `RunResult` instead of still showing a `skipped`?  Was able to backtrack somewhat to [`base.py`](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/task/base.py#L267) but this looks like its more around manifest creation. did we change the status's we accept in conversion?

Reply:
So this status is still `skipped` when running this in previous tests? 

Trying to understand what's going on here, `dependent_errors` model ref an ephemeral model that has issue? So it should be skipped vs error?

Reply:
hey @ChenyuLInx  thanks for taking a look. yes, it was skipped, I think I just solved this actually with a small change in how I was ref the models being used  for the class for testing. testing now in ci/cd.

Reply:
from what I can tell since we are getting an error from an invalid non existent method on the adapter via the `ephemeral_errors__base__base_copy_sql ` model I think we move it to skipped over raising an error as there would be no real clean error.